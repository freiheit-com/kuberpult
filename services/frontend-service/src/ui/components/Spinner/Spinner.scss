/*This file is part of kuberpult.

Kuberpult is free software: you can redistribute it and/or modify
it under the terms of the Expat(MIT) License as published by
the Free Software Foundation.

Kuberpult is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
MIT License for more details.

You should have received a copy of the MIT License
along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.

Copyright 2023 freiheit.com*/

.spinner {
    z-index: 10000;
    position: fixed;
    top: 50%;
    left: 50%;
    /* bring your own prefixes */
    transform: translate(-50%, -50%);
    .spinner-animation {
    }
    .spinner-message {
        @extend .sub-headline1;

        margin-top: 20px;
        background-color: white;
        box-shadow: 0 0 0 99999px rgba(0, 0, 0, 0.5);
        border-radius: 5px;
        text-align: center;
    }
    border-radius: 10px;
    background-color: transparent;
    color: black;
}
