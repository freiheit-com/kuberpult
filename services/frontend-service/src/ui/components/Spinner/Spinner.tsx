/*This file is part of kuberpult.

Kuberpult is free software: you can redistribute it and/or modify
it under the terms of the Expat(MIT) License as published by
the Free Software Foundation.

Kuberpult is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
MIT License for more details.

You should have received a copy of the MIT License
along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.

Copyright 2023 freiheit.com*/
import * as React from 'react';

import { PacmanLoader } from 'react-spinners';

export const Spinner: React.FC<{ message: string }> = (props): JSX.Element => {
    const { message } = props;
    return (
        <div className={'spinner'}>
            <div className={'spinner-animation'}>
                <PacmanLoader color="var(--mdc-theme-primary)" loading={true} size={100} speedMultiplier={1} />
            </div>
            <div className={'spinner-message'}>Loading {message}...</div>
        </div>
    );
};
