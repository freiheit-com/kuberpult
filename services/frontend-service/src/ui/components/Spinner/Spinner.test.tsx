/*This file is part of kuberpult.

Kuberpult is free software: you can redistribute it and/or modify
it under the terms of the Expat(MIT) License as published by
the Free Software Foundation.

Kuberpult is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
MIT License for more details.

You should have received a copy of the MIT License
along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.

Copyright 2023 freiheit.com*/
import { render } from '@testing-library/react';
import React from 'react';
import { Spinner } from './Spinner';

const cases: { name: string }[] = [
    {
        name: 'Renders',
    },
];

describe('Spinner', () => {
    const getNode = () => <Spinner message={'hello'} />;
    const getWrapper = () => render(getNode());

    describe.each(cases)('Renders', (testcase) => {
        it(testcase.name, () => {
            //given
            // when
            const { container } = getWrapper();
            // then
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
