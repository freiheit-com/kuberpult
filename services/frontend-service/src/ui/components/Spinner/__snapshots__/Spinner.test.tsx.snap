// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders Renders 1`] = `
<div
  class="spinner"
>
  <div
    class="spinner-animation"
  >
    <span
      style="display: inherit; position: relative; font-size: 0px; height: 200px; width: 200px;"
    >
      <span
        style="width: 0px; height: 0px; border-right: 100px solid transparent; border-top: 100px solid transparent; border-radius: 100px; position: absolute; animation: react-spinners-PacmanLoader-pacman-1 0.8s infinite ease-in-out; animation-fill-mode: both;"
      />
      <span
        style="width: 0px; height: 0px; border-right: 100px solid transparent; border-bottom: 100px solid transparent; border-radius: 100px; position: absolute; animation: react-spinners-PacmanLoader-pacman-2 0.8s infinite ease-in-out; animation-fill-mode: both;"
      />
      <span
        style="width: 33.333333333333336px; height: 33.333333333333336px; margin: 2px; border-radius: 100%; transform: translate(0, -25px); position: absolute; top: 100px; left: 400px; animation: react-spinners-PacmanLoader-ball 1s 0.5s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="width: 33.333333333333336px; height: 33.333333333333336px; margin: 2px; border-radius: 100%; transform: translate(0, -25px); position: absolute; top: 100px; left: 400px; animation: react-spinners-PacmanLoader-ball 1s 0.75s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="width: 33.333333333333336px; height: 33.333333333333336px; margin: 2px; border-radius: 100%; transform: translate(0, -25px); position: absolute; top: 100px; left: 400px; animation: react-spinners-PacmanLoader-ball 1s 1s infinite linear; animation-fill-mode: both;"
      />
      <span
        style="width: 33.333333333333336px; height: 33.333333333333336px; margin: 2px; border-radius: 100%; transform: translate(0, -25px); position: absolute; top: 100px; left: 400px; animation: react-spinners-PacmanLoader-ball 1s 1.25s infinite linear; animation-fill-mode: both;"
      />
    </span>
  </div>
  <div
    class="spinner-message"
  >
    hello
    ...
  </div>
</div>
`;
